### PR TITLE
feat(core): tooltip show hide delay

### LINKED ISSUE #202: [FEATURE] PointerHint add Delay Input
# 🚀 Feature request

Add a delay input for the PointerHint, so that it can be used similiar to a tooltip, but following the cursor
https://taiga-ui.dev/tui-pointer-hint

### PR DESCRIPTION
**Pull request by KarimovDev**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Closes #202 

## What is the new behavior?

Copy of #205 with implementation of review notes

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

**Review by waterplea — CHANGES_REQUESTED** [review]
Small comments. @MarsiBarsi please take a look at this PR as well. Do you think hint controller has to be involved as well?

File: projects/core/directives/pointer-hint/pointer-hint.directive.ts [hunk 1]
```diff
@@ -48,18 +64,27 @@ export class TuiPointerHintDirective extends AbstractTuiHint {
...
 
-        hint$.pipe(takeUntil(destroy$)).subscribe({
-            next: visible => {
-                if (visible) {
-                    this.showTooltip();
-                } else {
+        hint$
+            .pipe(
+                switchMap(visible => {
+                    return of(visible).pipe(
```
[the remark this answers is not in the document]
Comment (KarimovDev) [reply projects/core/directives/pointer-hint/pointer-hint.directive.ts]: Done

File: projects/demo/src/modules/components/tooltip/tooltip.component.ts [hunk 2]
```diff
@@ -61,4 +61,8 @@ export class ExampleTuiTooltipComponent {
     direction: TuiDirection = this.directionVariants[2];
 
     describeId = '';
+
+    showDelay = 0;
```
[the remark this answers is not in the document]
Comment (KarimovDev) [reply projects/demo/src/modules/components/tooltip/tooltip.component.ts]: You are right, I've fixed it

**Comment by KarimovDev** [comment]
> Small comments. @MarsiBarsi please take a look at this PR as well. Do you think hint controller has to be involved as well?

I think It has sense. I've added delay settings to hint controller

**Comment by MarsiBarsi** [comment]
Hey @KarimovDev thanks for your PR and mentioning me. I'll take a look today

**Review by MarsiBarsi — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**